crates/anvil/src/eth/backend/executor.rs

```diff
@@ -284,9 +285,12 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
         let nonce = account.nonce;
 
         // records all call and step traces
-        let mut inspector = Inspector::default().with_tracing();
-        if self.enable_steps_tracing {
-            inspector = inspector.with_steps_tracing();
+        let mut inspector = Inspector::default();
+        if !self.disable_tracing {
+            inspector = inspector.with_tracing();
+            if self.enable_steps_tracing {
+                inspector = inspector.with_steps_tracing();
+            }
```

Comment:
we can leave the setup as is, but if tracing is disabled we need to change this:

foundry_evm::utils::new_evm_with_inspector(&mut *self.db, env, &mut inspector);

to evm setup without inspector, which is always faster than with any inspector